cardano-db/src/Cardano/Db/Multiplex.hs

```diff
@@ -85,7 +85,7 @@ runExtraMigrations trce blockNoDiff consumed pruned = do
       liftIO $ logInfo trce "Extra migration consumed_tx_out already executed"
     (True, False, False) -> do
       liftIO $ logError trce migratedButNotSet
-      panic migratedButNotSet
+      liftIO $ throwIO $ userError (show migratedButNotSet)
```

Comment:
These lines look a lot like `logAndThrowIO` except it's not IO. See comment on `logAndThrowIO` on how to reuse it more.